airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -501,18 +343,31 @@ def _fetch_next_page(
         stream_state: Optional[Mapping[str, Any]] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Tuple[requests.PreparedRequest, requests.Response]:
-        request = self._create_prepared_request(
-            path=self.path(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
+
+        request, response = self._http_client.send_request(
```

Comment:
> How will we handle the migration for that?

I don't see any way to gracefully migrate this behaviour, because `httpClient` may raise `MessageRepresentationAirbyteTracedErrors(AirbyteTracedException)`, while we are waiting for `httperror` (which has .response property).

> I fear it'll break some of the connectors like source-amazon-ads because of [this](https://github.com/airbytehq/airbyte/blob/86ddeb0a5dfd1c14581f1f0841f6cf42d8edcbec/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py#L116-L123)

you're right, for `source-amazon-ads` this will be a breaking change (for `source-github` as well).
This is expected, and we'll need to migrate our code by setting `errorhandlers` explicitly.

Reply:
Should we document this [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/cdk-migrations.md) and have a plan to update these? I think we should at least update the ones that will be added in the CI next week which would be source-shopify, source-facebook-marketing and source-zendesk-support. I don't mind helping on that. source-shopify was already partially migrated to the error handler so the change might be smaller for this one

Reply:
> Should we document this [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/cdk-migrations.md) and have a plan to update these?

Sure, I'll update Migration Guide.
Should we consider these changes as breaking and bump CDK version to `3.0` ? I agreed that they are breaking, but only for specific cases.

I can update `shopify` and `zendesk-support` after that, migration should be smooth here (source-facebook-marketing does not use `httpstream`, this source is based upon facebook business py library).

Reply:
I would assume that this would be a version 3.0.0 in that case yes because we purposely break the interface but I'm curious what @girarda might think

Reply:
Yes let's flag this as a breaking change